Model/Ui/CapabilitiesConfigProvider.php

```diff
@@ -0,0 +1,47 @@
+<?php
+namespace Omise\Payment\Model\Ui;
+
+use Magento\Checkout\Model\ConfigProviderInterface;
+use Magento\Payment\Api\PaymentMethodListInterface;
+use Magento\Store\Model\StoreManagerInterface;
+use Omise\Payment\Model\Capabilities;
+use Omise\Payment\Model\Config\Installment as OmiseInstallmentConfig;
+
+class CapabilitiesConfigProvider implements ConfigProviderInterface
+{
+    private $_storeManager;
+
+    /**
+     * @var Magento\Payment\Api\PaymentMethodListInterface;
+     */
+    private $_paymentLists;
+
+    public function __construct(
+        Capabilities               $capabilities,
+        PaymentMethodListInterface $paymentLists,
+        StoreManagerInterface      $storeManager
+    )
+    {
+        $this->capabilities    = $capabilities;
+        $this->_paymentLists   = $paymentLists;
+        $this->_storeManager   = $storeManager;
+    }
+
+    /**
+     * Retrieve assoc array of checkout configuration
+     *
+     * @return array
+     */
+    public function getConfig()
+    {
+        $listOfActivePaymentMethods = $this->_paymentLists->getActiveList($this->_storeManager->getStore()->getId());
+        foreach ($listOfActivePaymentMethods as $method) {
+            if ($method->getCode() === OmiseInstallmentConfig::CODE) {
+                return [ 
+                    'installment_backends' => $this->capabilities->retrieveInstallmentBackends()
```

Comment:
just want to note it out that we are sending out 3 unnecessary properties. From Capabilities API response,
```json
{
  "installment_bay": {
    "type": "installment",
    "currencies": [
      "thb"
    ],
    "allowed_installment_terms": [
      3,
      4,
      6,
      9,
      10
    ],
    "amount": {
      "min": 2000,
      "max": 100000000
    }
  }
}
```

In this pull request, it uses only `allowed_installment_terms` and `_id` (special attribute generated by Omise-PHP).
We don't use `type`, `currencies`, and `amount` at the moment.

Would be nice if we limit an access to only what we will use in the frontend.

Reply:
will use them in the future... when we will have correct values in amount.
than I will also use currencies, now, it is also hardcoded.